modules/core/include/opencv2/core/hal/intrin_rvv.hpp

```diff
@@ -584,63 +587,84 @@ OPENCV_HAL_IMPL_RVV_SELF_REINTERPRET(int64x2, s64)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint16x8, ushort, int64x2, int64, vuint16m1_t, vint64m1_t, u16, s64, u, i, 16, 64, 8, 2)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint32x4, uint, int8x16, schar, vuint32m1_t, vint8m1_t, u32, s8, u, i, 32, 8, 4, 16)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint32x4, uint, int16x8, short, vuint32m1_t, vint16m1_t, u32, s16, u, i, 32, 16, 4, 8)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint32x4, uint, int64x2, int64, vuint32m1_t, vint64m1_t, u32, s64, u, i, 32, 64, 4, 2)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint64x2, uint64, int8x16, schar, vuint64m1_t, vint8m1_t, u64, s8, u, i, 64, 8, 2, 16)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint64x2, uint64, int16x8, short, vuint64m1_t, vint16m1_t, u64, s16, u, i, 64, 16, 2, 8)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint64x2, uint64, int32x4, int, vuint64m1_t, vint32m1_t, u64, s32, u, i, 64, 32, 2, 4)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint8x16, uchar, float32x4, float, vuint8m1_t, vfloat32m1_t, u8, f32, u, f, 8, 32, 16, 4)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint16x8, ushort, float32x4, float, vuint16m1_t, vfloat32m1_t, u16, f32, u, f, 16, 32, 8, 4)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint64x2, uint64, float32x4, float, vuint64m1_t, vfloat32m1_t, u64, f32, u, f, 64, 32, 2, 4)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(int8x16, schar, float32x4, float, vint8m1_t, vfloat32m1_t, s8, f32, i, f, 8, 32, 16, 4)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(int16x8, short, float32x4, float, vint16m1_t, vfloat32m1_t, s16, f32, i, f, 16, 32, 8, 4)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(int64x2, int64, float32x4, float, vint64m1_t, vfloat32m1_t, s64, f32, i, f, 64, 32, 2, 4)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint8x16, uchar, float64x2, double, vuint8m1_t, vfloat64m1_t, u8, f64, u, f, 8, 64, 16, 2)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint16x8, ushort, float64x2, double, vuint16m1_t, vfloat64m1_t, u16, f64, u, f, 16, 64, 6, 2)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(uint32x4, uint, float64x2, double, vuint32m1_t, vfloat64m1_t, u32, f64, u, f, 32, 64, 4, 2)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(int8x16, schar, float64x2, double, vint8m1_t, vfloat64m1_t, s8, f64, i, f, 8, 64, 16, 2)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(int16x8, short, float64x2, double, vint16m1_t, vfloat64m1_t, s16, f64, i, f, 16, 64, 8, 2)
+OPENCV_HAL_IMPL_RVV_TWO_TIMES_REINTERPRET(int32x4, int, float64x2, double, vint32m1_t, vfloat64m1_t, s32, f64, i, f, 32, 64, 4, 2)
+
+// Three times reinterpret
+inline v_float32x4 v_reinterpret_as_f32(const v_float64x2& v) \
+{ \
+    return v_float32x4(vreinterpret_v_u32m1_f32m1(vreinterpret_v_u64m1_u32m1(vreinterpret_v_f64m1_u64m1(v))));\
+} \
+inline v_float64x2 v_reinterpret_as_f64(const v_float32x4& v) \
+{ \
+    return v_float64x2(vreinterpret_v_u64m1_f64m1(vreinterpret_v_u32m1_u64m1(vreinterpret_v_f32m1_u32m1(v))));\
+}
```

Comment:
Sounds very suspicious. I think `vfncvt` call fits better.

Reply:
I think `vfncvt` is more like type conversion but not reinterpret and can not use in here.

For example: 

When `v_float32x4 v_reinterpret_as_f32(const v_float64x2& v)` is called. We want to treat the 128bits(64 bit * 2) in the register as (32bits *4).

However, `vfloat32m1_t vfncvt_f_f_w_f32m1 (vfloat64m2_t src, size_t vl);` means there are four float64 is converting to four float32.